app/models/course.rb

```diff
@@ -148,4 +148,31 @@ def funding
       'fee'
     end
   end
+
+  def publish_sites
+    update_new_to_running
+    update_published_if_running
+    # as update_all is called and doesn't trigger a save this has to be manually tested
+    update(changed_at: Time.now.utc)
+  end
+
+  def publish_enrichment(current_user_id)
```

Comment:
I feel like, assuming you'll always have it, the user object should be here instead of the id.